homeassistant/components/motion_blinds/cover.py

```diff
@@ -0,0 +1,278 @@
+"""Support for Motion Blinds using their WLAN API."""
+
+import logging
+
+from motionblinds import BlindType
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    DEVICE_CLASS_AWNING,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASS_CURTAIN,
+    DEVICE_CLASS_GATE,
+    DEVICE_CLASS_SHADE,
+    DEVICE_CLASS_SHUTTER,
+    CoverEntity,
+)
+
+from .const import DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Motion Blind from a config entry."""
+    entities = []
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id]
+    for blind in motion_gateway.device_list.values():
+        await hass.async_add_executor_job(blind.Update)
+
+        if blind.type in [
+            BlindType.RollerBlind,
+            BlindType.RomanBlind,
+            BlindType.HoneycombBlind,
+            BlindType.DimmingBlind,
+            BlindType.DayNightBlind,
+        ]:
+            entities.append(
+                MotionPositionDevice(blind, DEVICE_CLASS_SHADE, config_entry)
+            )
+
+        elif blind.type in [
+            BlindType.RollerShutter,
+            BlindType.Switch,
+        ]:
+            entities.append(
+                MotionPositionDevice(blind, DEVICE_CLASS_SHUTTER, config_entry)
+            )
+
+        elif blind.type in [BlindType.RollerGate]:
+            entities.append(
+                MotionPositionDevice(blind, DEVICE_CLASS_GATE, config_entry)
+            )
+
+        elif blind.type in [BlindType.Awning]:
+            entities.append(
+                MotionPositionDevice(blind, DEVICE_CLASS_AWNING, config_entry)
+            )
+
+        elif blind.type in [
+            BlindType.Curtain,
+            BlindType.CurtainLeft,
+            BlindType.CurtainRight,
+        ]:
+            entities.append(
+                MotionPositionDevice(blind, DEVICE_CLASS_CURTAIN, config_entry)
+            )
+
+        elif blind.type in [
+            BlindType.VenetianBlind,
+            BlindType.ShangriLaBlind,
+        ]:
+            entities.append(MotionTiltDevice(blind, DEVICE_CLASS_BLIND, config_entry))
+
+        elif blind.type in [BlindType.DoubleRoller]:
+            entities.append(MotionTiltDevice(blind, DEVICE_CLASS_SHADE, config_entry))
+
+        elif blind.type in [BlindType.TopDownBottomUp]:
+            entities.append(
+                MotionTDBUDevice(blind, DEVICE_CLASS_SHADE, config_entry, "Top")
+            )
+            entities.append(
+                MotionTDBUDevice(blind, DEVICE_CLASS_SHADE, config_entry, "Bottom")
+            )
+
+        else:
+            _LOGGER.warning("Blind type '%s' not yet supported", blind.blind_type)
+
+    async_add_entities(entities)
+
+
+class MotionPositionDevice(CoverEntity):
```

Comment:
In principle yes, however I have experianced that the device status is not properly represented in the API, for my blind the self._blind.status almost always returns "stopped" even if it is moving up or down, moreover the position of the blind is not updated during movement of the blind, the gateway still gives the last position, only if the blind stops moving the gateway gives the new position.

Therefore I chose to not implement is_opening and is_closing, maybe this will be possible in the future if push is implemented instead of poll.
But I first want to get this integration in HomeAssistant before I start work on push implementation.